.github/workflows/links.yml

```diff
@@ -0,0 +1,20 @@
+name: links
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+  check-links:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: lycheeverse/lychee-action@v1.5.1
```

Comment:
(This is more of observation , then an action for change) 
opentelemetry-collector-contrib  is using https://github.com/tcort/markdown-link-check , https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/eb42062694639daeed9a6b099c1d34cd8a7d9278/.github/workflows/check-links.yaml#L37 

opentelemetry-demo app is using https://github.com/open-telemetry/opentelemetry-demo/blob/d44100d0390a227475928e38239db8cc628215cc/.github/workflows/checks.yml#L51

opentelemetry-go is using `lycheeverse/lychee-action`  https://github.com/open-telemetry/opentelemetry-go/blob/11339775566920f5caa82866eb0b6b291ede9df9/.github/workflows/links.yml#L19 

I don't have experience with all of them, but it would have been better if we had consistency among projects.

Reply:
yup , agree `lychee` looks good